trie/stacktrie.go

```diff
@@ -428,10 +435,9 @@ func (st *StackTrie) hashRec(hasher *hasher) {
 		encodedNode = hasher.encodedBytes()
 
 	case extNode:
-		st.children[0].hashRec(hasher)
+		st.children[0].hashRec(hasher, append(path, st.key...))
 
-		sz := hexToCompactInPlace(st.key)
-		n := rawShortNode{Key: st.key[:sz]}
+		n := rawShortNode{Key: hexToCompact(st.key)}
```

Comment:
`hexToCompactInPlace` is replaced by `hexToCompact` for not changing the `st.key`. Otherwise the path prefix of children will be affected since they are sharing the same byte slice.